src/index.d.ts

```diff
@@ -191,34 +191,17 @@ declare namespace preact {
 	 * Global options for preact
 	 */
 	interface Options {
-		/** Attach a hook that is invoked before render, mainly to check the arguments. */
-		root?(vnode: ComponentChild, parent: Element | Document | ShadowRoot | DocumentFragment): void;
 		/** Attach a hook that is invoked whenever a VNode is created. */
-		vnode(vnode: VNode): void;
-		/** Attach a hook that is invoked after a tree was mounted or was updated. */
-		commit?(vnode: VNode): void;
-		/** Attach a hook that is invoked immediately before a component is unmounted. */
+		vnode?(vnode: VNode): void;
+		/** Attach a hook that is invoked immediately before a vnode is unmounted. */
 		unmount?(vnode: VNode): void;
 		/** Attach a hook that is invoked before a vnode is diffed. */
 		diff?(vnode: VNode): void;
```

Comment:
I'm a little torn on this one. I'm not sure how it could be used, but it feels complimentary to `diffed`. Perhaps we should leave it out until we have a reason not to? Not sure...

Reply:
Yeah, this hook is tough. Wondering if most of it could be accomplished by mutating a `vnode` in `options.vnode()` instead. This one is mainly needed for profiling purposes as we need to have a marker when the node actually starts and finishes rendering.

Reply:
I went ahead and marked it as private. If we want to bring it back it's as simple as reverting the latest commit, so just let me know.